### PR TITLE
test(react/vanilla-utils/freezeAtom): add tests for null and primitive handling

### PR DESCRIPTION
**Pull request by sukvvon**
## Related Bug Reports or Discussions

Fixes #

### before

image

### after

image

## Check List

- [x] `pnpm run fix` for formatting and linting code and docs

### CODE REVIEW

File: tests/react/vanilla-utils/freezeAtom.test.tsx [hunk 1]
```diff
@@ -35,6 +35,64 @@ it('freezeAtom basic test', async () => {
   await screen.findByText('isFrozen: true')
 })
 
+it('freezeAtom handles null correctly', async () => {
+  const nullAtom = atom<null, [null], void>(null, (_get, set, _arg) => {
```
Comment (dai-shi) [inline tests/react/vanilla-utils/freezeAtom.test.tsx:39]: Does this work?
```suggestion
  const nullAtom = atom(null, (_get, set, _arg) => {
```

Comment (dai-shi) [reply tests/react/vanilla-utils/freezeAtom.test.tsx:39]: Oh, I see. because the arg is required.
Let's do this then:
```suggestion
  const nullAtom = atom(null, (_get, set, _arg: null) => {
```

File: tests/react/vanilla-utils/freezeAtom.test.tsx [hunk 2]
```diff
@@ -35,6 +35,64 @@ it('freezeAtom basic test', async () => {
...
+  )
+
+  await screen.findByText('value is null: true')
+
+  await userEvent.click(screen.getByText('set null'))
+  await screen.findByText('value is null: true')
+})
+
+it('freezeAtom handles primitive correctly', async () => {
+  const numberAtom = atom<number, [number], void>(123, (_get, set) => {
```
Comment (sukvvon) [inline tests/react/vanilla-utils/freezeAtom.test.tsx:66]:
```suggestion
  const numberAtom = atom(123, (_get, set, _arg: number) => {
```

@dai-shi How about this?

Comment (dai-shi) [reply tests/react/vanilla-utils/freezeAtom.test.tsx:66]: this too.

File: tests/react/vanilla-utils/freezeAtom.test.tsx [hunk 3]
```diff
@@ -35,6 +35,64 @@ it('freezeAtom basic test', async () => {
...
+    set(numberAtom, 456)
+  })
+
+  const Component = () => {
+    const [value, setValue] = useAtom(freezeAtom(numberAtom))
+    return (
+      <>
+        <button onClick={() => setValue(456)}>set number</button>
+        <div>value is frozen: {`${Object.isFrozen(value)}`}</div>
```
Comment (dai-shi) [inline tests/react/vanilla-utils/freezeAtom.test.tsx:75]: I'm not sure what this is testing. Does this test pass even if we omit `freezeAtom`?

Comment (sukvvon) [reply tests/react/vanilla-utils/freezeAtom.test.tsx:75]: @dai-shi 

Thanks for the feedback. I missed the fact that freeze doesn’t really apply to primitives. I’ll revise the test to make its purpose clearer.

File: tests/react/vanilla-utils/freezeAtom.test.tsx [hunk 4]
```diff
@@ -35,6 +35,62 @@ it('freezeAtom basic test', async () => {
...
+    <StrictMode>
+      <Component />
+    </StrictMode>,
+  )
+
+  await screen.findByText('value is null: true')
+
+  await userEvent.click(screen.getByText('set null'))
+
+  await screen.findByText('value is null: true')
```
Comment (dai-shi) [inline tests/react/vanilla-utils/freezeAtom.test.tsx:63]: Should we set something not null too?

Comment (sukvvon) [reply tests/react/vanilla-utils/freezeAtom.test.tsx:63]: > Should we set something not null too?

@dai-shi 

I think this should be good enough as is, but feel free to let me know if you think anything else should be added!

Comment (sukvvon) [inline tests/react/vanilla-utils/freezeAtom.test.tsx:63]:
```suggestion
```

File: tests/react/vanilla-utils/freezeAtom.test.tsx [hunk 5]
```diff
@@ -35,6 +35,58 @@ it('freezeAtom basic test', async () => {
...
+it('freezeAtom handles primitive correctly', async () => {
+  const numberAtom = atom(123, (_get, set, _arg: number) => {
+    set(numberAtom, 456)
+  })
+
+  const Component = () => {
+    const [value, setValue] = useAtom(freezeAtom(numberAtom))
+    return (
+      <>
+        <button onClick={() => setValue(456)}>set number</button>
```
Comment (sukvvon) [inline tests/react/vanilla-utils/freezeAtom.test.tsx:71]:
```suggestion
  const numberAtom = atom(123, (_get, set, _ignored: number) => {
    set(numberAtom, 456)
  })

  const Component = () => {
    const [value, setValue] = useAtom(freezeAtom(numberAtom))
    return (
      <>
        <button onClick={setValue}>set number</button>
```
@dai-shi 

How about aligning the format with the existing test code for consistency?

File: tests/react/vanilla-utils/freezeAtom.test.tsx [hunk 6]
```diff
@@ -35,6 +35,58 @@ it('freezeAtom basic test', async () => {
...
+    </StrictMode>,
+  )
+
+  await screen.findByText('value is null: true')
+})
+
+it('freezeAtom handles primitive correctly', async () => {
+  const numberAtom = atom(123, (_get, set, _arg: number) => {
+    set(numberAtom, 456)
+  })
```
[the remark this answers is not in the document]
Comment (sukvvon) [reply tests/react/vanilla-utils/freezeAtom.test.tsx:65]: @dai-shi 

The intention was to maintain consistency with the existing test code style

File: tests/react/vanilla-utils/freezeAtom.test.tsx [hunk 7]
```diff
@@ -35,6 +35,56 @@ it('freezeAtom basic test', async () => {
   await screen.findByText('isFrozen: true')
 })
 
+it('freezeAtom handles null correctly', async () => {
+  const nullAtom = atom(null, (_get, set, _arg: null) => {
+    set(nullAtom, null)
+  })
```
Comment (sukvvon) [inline tests/react/vanilla-utils/freezeAtom.test.tsx:41]:
```suggestion
  const nullAtom = atom(null)
```

**Comment by sukvvon** [comment]
> Let's fix the existing tests:
> 
> ```diff
> diff --git a/tests/react/vanilla-utils/freezeAtom.test.tsx b/tests/react/vanilla-utils/freezeAtom.test.tsx
> index ca135bf..11169bb 100644
> --- a/tests/react/vanilla-utils/freezeAtom.test.tsx
> +++ b/tests/react/vanilla-utils/freezeAtom.test.tsx
> @@ -7,17 +7,16 @@ import { atom } from 'jotai/vanilla'
>  import { freezeAtom, freezeAtomCreator } from 'jotai/vanilla/utils'
>  
>  it('freezeAtom basic test', async () => {
> -  const objAtom = atom({ deep: {} }, (_get, set, _ignored?) => {
> -    set(objAtom, { deep: {} })
> -  })
> +  const objAtom = atom({ deep: { count: 0 } })
>  
>    const Component = () => {
>      const [obj, setObj] = useAtom(freezeAtom(objAtom))
>      return (
>        <>
> -        <button onClick={setObj}>change</button>
> +        <button onClick={() => setObj({ deep: { count: 1 } })}>change</button>
>          <div>
> -          isFrozen: {`${Object.isFrozen(obj) && Object.isFrozen(obj.deep)}`}
> +          count: {obj.deep.count}, isFrozen:{' '}
> +          {`${Object.isFrozen(obj) && Object.isFrozen(obj.deep)}`}
>          </div>
>        </>
>      )
> @@ -29,10 +28,10 @@ it('freezeAtom basic test', async () => {
>      </StrictMode>,
>    )
>  
> -  await screen.findByText('isFrozen: true')
> +  await screen.findByText('count: 0, isFrozen: true')
>  
>    await userEvent.click(screen.getByText('change'))
> -  await screen.findByText('isFrozen: true')
> +  await screen.findByText('count: 1, isFrozen: true')
>  })
>  
>  describe('freezeAtomCreator', () => {
> ```

@dai-shi 

I reflected it.

**Review by dai-shi — APPROVED** [review]
LGTM
